File: source/development/device-api/device-server-writing/writing_java/StepperMotor3.java
public int dev_read_direction(int axis) throws DevFailed
{
	if (axis < 0 || axis > SM_MAX_MOTORS)
	{
		Util.out1.println("Steppermotor.dev_read_direction(): axis out of range !");
		
		StringBuffer o = new StringBuffer("Axis number ");
		o.append(axis);
		o.append(" out of range");
		
		Except.throw_exception("StepperMotor_AxisOutOfRange",
				       o.toString(),
				       "StepperMotor.dev_read_direction()");
	}

	return direction[axis];
}

public boolean direct_cmd_allowed(Any data_in)
{
	Util.out2.println("In StepperMotor.direct_cmd_allowed method");

	return true;
}

File: source/development/device-api/device-server-writing/writing/steppermotor3.cpp
long StepperMotor::dev_read_direction(long axis)
{
	if (axis < 0 || axis > AGSM_MAX_MOTORS)
	{
		cout1 << "Steppermotor::dev_read_direction(): axis out of range !";
		cout1 << endl;
		TangoSys_OMemStream o;
		
		o << "Axis number " << axis << " out of range" << ends;
		throw_exception((const char *)"StepperMotor_OutOfRange",
				o.str(),
				(const char *)"StepperMotor::dev_read_direction");
	}

	return direction[axis];
}


bool StepperMotor::direct_cmd_allowed(const CORBA::Any &in_data)
{
	cout2 << "In direct_cmd_allowed() method" << endl;
	
	return true;
}

